src/nethsec/inventory/__init__.py

```diff
@@ -446,3 +446,68 @@ def fact_ddns(uci: EUci):
 def fact_snmp (uci: EUci):
     snmp = _run_status("/etc/init.d/snmpd running")
     return { 'enabled': snmp == 0 }
+
+def fact_wiregard(uci: EUci):
+    ret = { 'instances': 0, 'statistics':[] }
+    wg= []
+    user_db = {}
+    interfaces = utils.get_all_by_type(uci, "network", "interface")
+    for i in interfaces:
+        interface = interfaces[i]
+        if interface.get("proto") == "wireguard":
+           wg.append(i)
+           ret['instances'] += 1
+           user_db.update({i:'main'})
+           if interface.get("ns_user_db"):
+               user_db.update({i:interface.get('ns_user_db')})
+    ## iterate over all wireguard interfaces from wg
+    ## find the number of peers from wireguard_wg1
+    for w in wg:
+        peers = utils.get_all_by_type(uci, "network", 'wireguard_'+w)
+        ret['statistics'].append({'server': w, "peers": len(peers), "ns_user_db": user_db[w]})
+    return ret
+
+
+def fact_snort(uci: EUci):
+    ret = { 'enabled': False, 'policy': '', 'oink_enabled': False, 'disabled_rules': 0, 'bypass_src_ipv4': 0, 'bypass_src_ipv6': 0, 'bypass_dst_ipv4': 0, 'bypass_dst_ipv6': 0 }
+
+    ret['enabled'] = uci.get('snort', 'snort', 'enabled', default='0') == '1'
+    ret['policy'] = uci.get('snort', 'snort', 'ns_policy', default='')
+    ret['oink_enabled'] = True if uci.get('snort', 'snort', 'oinkcode', default='') else False
+
+    # count list of ns_disabled_rules
+    try :
+        ret['disabled_rules'] = len(uci.get_all('snort', 'snort', 'ns_disabled_rules'))
+    except:
+        pass
```

Comment:
You can even let Pyuci do the heavy lifting for you, this won't need any try catch
```suggestion
    ret['disabled_rules'] = len(uci.get('snort', 'snort', 'ns_disabled_rules', list=True, default=[]))
```